src/data/repositories/D2ApiMetadata.ts

```diff
@@ -65,60 +65,63 @@ export class D2ApiConfig {
 
     private getMetadata(): FutureData<D2Config> {
         return apiToFuture(this.api.metadata.get(metadataFields)).map(d2Response => {
+            const getOrThrowMetadata = (metadataKey: keyof typeof metadataFields, code: string) =>
+                getOrThrow(d2Response[metadataKey], code);
+
             return {
                 attributes: this.buildAttributes(d2Response.attributes),
                 categories: {
-                    project: getOrThrow(d2Response.categories, metadataCodes.categories.project),
+                    project: getOrThrowMetadata("categories", metadataCodes.categories.project),
                 },
                 dataElementGroupSets: {
-                    coreCompetency: getOrThrow(
-                        d2Response.dataElementGroupSets,
+                    coreCompetency: getOrThrowMetadata(
+                        "dataElementGroupSets",
                         metadataCodes.dataElementGroupSets.coreCompetency
```

Comment:
Not single-line, but yes, it may hinder readibility if we push it too much. let's keep as is.

(other strategies: move to separate function to reduce indentation level, create shorter name for helper functions, rename to shorter names the variables in the block)